internal/controller/standalone_pgadmin/service.go

```diff
@@ -0,0 +1,115 @@
+/*
+ Copyright 2023 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package standalone_pgadmin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/crunchydata/postgres-operator/internal/initialize"
+	"github.com/crunchydata/postgres-operator/internal/naming"
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+)
+
+// +kubebuilder:rbac:groups="",resources="services",verbs={get}
+// +kubebuilder:rbac:groups="",resources="services",verbs={create,delete,patch}
+
+// reconcilePGAdminService writes the Service that resolves to pgAdmin.
+func (r *PGAdminReconciler) reconcilePGAdminService(
+	ctx context.Context, pgadmin *v1beta1.PGAdmin,
+) (*corev1.Service, error) {
+	service, err := r.generatePGAdminService(pgadmin)
+
+	if err == nil {
+		err = errors.WithStack(r.setControllerReference(pgadmin, service))
+	}
+
+	if err == nil {
+		err = errors.WithStack(r.apply(ctx, service))
+	}
+	return service, err
+}
+
+// generatePGAdminService returns a v1.Service that exposes pgAdmin pods.
+func (r *PGAdminReconciler) generatePGAdminService(
```

Comment:
:pick: To keep things consistent, I'd ideally like this to be `service` or, if we're going to have more than one defined, perhaps `pgAdminService`?

Reply:
I agree in theory, but left it like this because the other functions we have that define K8s objects -- `pod`, `secret`, `configmap`, et al. -- are all _functions_, whereas this is a method that requires the reconciler. Why does it require the reconciler? Because (a) it's a copy of the original `generatePGAdminService` which (b) records an event in the case of someone trying to set the NodePort for a ClusterIP service type.

I went back and forth on removing that event recorder... actually, now that I'm talking it out, I could move this check to the parent function.

Reply:
Makes sense to me. I remember hitting some similar situations on my PR. If we could move that to the parent function, I think that'd be nice.